### ISSUE TITLE
False Positive: checkstyle:com.puppycrawl.tools.checkstyle.checks.annotation.AnnotationOnSameLineCheck   

### ISSUE THREAD

**Issue by alixwar** [checkstyle/checkstyle#5522]
**Checkstyle version**: 8.7
**Tokens**: CLASS_DEF,INTERFACE_DEF,ENUM_DEF,METHOD_DEF,CTOR_DEF,VARIABLE_DEF

This code triggers a violation for rule  "Annotation On Same Line":

```
    @SuppressWarnings("squid:S4142")
    private AccessRequestDto(Parcel in) {
        this(in.readString(), in.readString(), new AccessRequestScheduleDto(in), in.readString(), in.readString());
    }
```
---------------

It is not possible to follow the rule (code below does not compile, this must be the first statement in a constructor body):

```
    private AccessRequestDto(Parcel in) {
        @SuppressWarnings("squid:S4142")
        this(in.readString(), in.readString(), new AccessRequestScheduleDto(in), in.readString(), in.readString());
    }
```
--------------

Suggestion: Don't trigger if the line causing a violation starts with `this(`

**Comment by rnveach** [comment]
@alixwar 
> It is not possible to follow the rule (code below does not compile, this must be the first statement in a constructor body):

Check is not requesting you to put `SuppressWarnings` inside the constructor. It is requesting you put it on the same line as the constructor declaration `private AccessRequestDto(Parcel in) {`.

This doesn't look like a false positive. Check is behaving as expected as you have configured it.
I suggest you look over the documentation for this check. http://checkstyle.sourceforge.net/config_annotation.html#AnnotationOnSameLine

If you still have issues, report problem like http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug.3F .

````
$ cat TestClass.java
public class TestClass {
    @SuppressWarnings("squid:S4142")
    private AccessRequestDto(Parcel in) {
        this();
    }

    @SuppressWarnings("squid:S4142") private AccessRequestDto(Parcel in) {
        this();
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="AnnotationOnSameLineCheck" />
    </module>
</module>

$ java -jar checkstyle-8.7-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:2: Annotation 'SuppressWarnings' should be on the same line with its target. [AnnotationOnSameLine]
Audit done.
Checkstyle ends with 1 errors.
````

See there is no violation on line 7.

**Comment by alixwar** [comment]
@rnveach Thanks for your comment!

I guess that I want to configure the rule to allow a line break then since I don't want annotations on the same line (creates unnecessary violations of our character line limit)

**Comment by rnveach** [comment]
>  I want to configure the rule to allow a line break 

Have you tried `AnnotationLocation`? http://checkstyle.sourceforge.net/config_annotation.html#AnnotationLocation

**Comment by alixwar** [comment]
No, thanks for the pointer! :-)
